### ISSUE TITLE
"sum to zero" animation is obscured by item that drops down

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/equality-explorer#28]
In a request for feedback on [1.0.0-dev.48](https://www.colorado.edu/physics/phet/dev/html/equality-explorer/1.0.0-dev.48/equality-explorer_en.html), Argenta P. said via email:

> 2) I like the yellow halo effect for cancelling +/- numbers/variables. (And I don't mind it being a circle always.) If you're cancelling something that has another object on top, would it be possible to delay the animation of the top object falling until after the "0" has disappeared? I found it a bit confusing to have it drop first, because then it seemed like a 0 was fading into a 1. The 0 animation is nice, but if you can't do a delay, I think I would prefer the cancelled blocks to just disappear (or be "squashed" by the falling block, without the 0 animation at all.

I replied:

> Amanda and I had discussed this, and decided that an item should fall immediately when an item below it is removed.  See https://github.com/phetsims/equality-explorer/issues/17#issue-271032750: "when an item on the scale is replaced in this manner, items on the scale above it drop down immediately”.  
> 
> So it’s not currently possible (in the implementation) to delay the animation of the object falling down to fill a slot.  The animation can’t occur until the item is removed, and removing an item results in anything above it falling down - it’s an action that is totally independent of, and knows nothing about, the “0” animation).  Delaying the "drop" would also create some natty multi-touch issues.
> 
> Your other proposal (having the ‘0’ animation in some scenarios but not in others) seems conceptually problematic. I believe that we should show the animation consistently, or not at all.

**Comment by pixelzoom** [comment]
Via email, Argenta said:
> Regarding the 0 animation, that makes sense that it's separate from the drop and is complicated. My second suggestion was actually to remove the 0 animation in all cases (not just when there's an item above) if the drop couldn't be delayed. But if you all think the 0 animation is more important, that's fine too.

**Comment by pixelzoom** [comment]
@amanda-phet I think we can improve the clarity of the "sum to zero" animation by tweaking the animation parameters a bit.  

In [1.0.0-dev.51](https://www.colorado.edu/physics/phet/dev/html/equality-explorer/1.0.0-dev.51/equality-explorer_en.html), the initial opacity is 0.85 (slightly transparent) and the duration is 500 ms.  So the animation is already a little fuzzy when it starts (because it's a little transparent), and then it's over rather quickly.

[1.0.0-dev.52](https://www.colorado.edu/physics/phet/dev/html/equality-explorer/1.0.0-dev.52/equality-explorer_en.html), the initial opacity is 1.0 (opaque) and the duration is 750 ms. The animation starts off nice and crisp, because it's opaque.  It still occurs quickly, but a little less quickly so you can really see what's happening. 

In both cases, the rate of opacity change increases over time, so that most of the "fade out" happens at the end. 

I think 1.0.0-dev.52 is a nice improvement. But note that it is now slightly different than Expression Exchange. Imo, that's OK, since the context is different. 

Let me know if you approve of the change in 1.0.0-dev.52, or want to revert to the parameters used in 1.0.0-dev.51.

**Comment by amanda-phet** [comment]
I prefer the slightly transparent feel in dev.51 over the fully opaque look in dev.52, so overall I prefer the animation in dev.51.

**Comment by pixelzoom** [comment]
What about the slightly longer animation time in dev.52?

**Comment by amanda-phet** [comment]
I almost asked if you could do a combo of 0.85 initial transparency and 750ms animation time, but I wasn't sure that made much sense since it doesn't have as far to go in terms of disappearing. Should we try that?

**Comment by pixelzoom** [comment]
I think it makes sense.   When an item drops down and the animation occurs, that's a lot of stuff happening at once.  A longer animation time gives the user more time to process what's happening.

**Comment by amanda-phet** [comment]
I'm happy with the latest tweaks (longer fade time, 0.85 starting opacity for the glow, 1.0 starting opacity for the "0"). Thanks!

**Comment by pixelzoom** [comment]
Unassigning for now. We will revisit after @amanda-phet has done informal interviews on 1/8/18.

**Comment by amanda-phet** [comment]
The students in interviews didn't use the sum to zero functionality very much. One student didn't use it until prompted, the other accidentally found it. In both cases, students understood the meaning but didn't further explore it in complicated cases. 

This github issue was about the specific situation where an object drops down and is obscured by a zero pair, so if we want to know about how a student would interpret this particular scenario I might need to set it up and prompt a student to do that exact action. In the typical use of the sim as observed in interviews, I didn't notice any issues relevant to this issue. 

I think that the tweaks we made are an improvement to the feature, so I am ok closing this issue. If we want to test it as I described above, we can leave this open until the next round of interviews.
